custom/icds_reports/management/commands/migrate_tables.py

```diff
@@ -0,0 +1,158 @@
+from __future__ import absolute_import, print_function
+from __future__ import unicode_literals
+
+import argparse
+import inspect
+import logging
+import subprocess
+import sys
+from datetime import datetime, date
+
+import csv342 as csv
+from six.moves import input
+
+logger = logging.getLogger(__name__)
+
+
+def parse_date(s, default=None):
+    if not s:
+        return default
+
+    try:
+        return datetime.strptime(s, "%Y-%m-%d").date().replace(day=1)
+    except ValueError:
+        msg = "Not a valid date: '{0}'.".format(s)
+        raise argparse.ArgumentTypeError(msg)
+
+
+def run_migration(
+    table_path, source_db,
+    target_db, target_host, target_user,
+    start_date, end_date,
+    only_table, confirm, dry_run
+):
+    with open(table_path, 'r') as file:
+        tables = [
+            (source_table, parse_date(date_str, date.max), target_table)
+            for source_table, date_str, target_table in csv.reader(file)
+        ]
+
+    filtered_tables = filter_tables_by_date(tables, start_date, end_date)
+    if only_table:
+        filtered_tables = [
+            table for table in filtered_tables if table[0] == only_table
+        ]
+    if not filtered_tables:
+        raise Exception("No table to migrate")
+
+    if dry_run or _confirm('Preparing to migrate {} tables.'.format(len(filtered_tables))):
+        migrate_tables(
+            filtered_tables, source_db, target_db,
+            target_host, target_user, confirm, dry_run
+        )
+
+
+def migrate_tables(tables, source_db, target_db, target_host, target_user, confirm, dry_run):
+    commands = get_dump_load_commands(tables, source_db, target_db, target_host, target_user)
+    for source_table, target_table, cmd_parts in commands:
+        print(' '.join(cmd_parts))
+        if not dry_run and (not confirm or _confirm('Migrate {} to {}'.format(source_table, target_table))):
+            code = subprocess.call(cmd_parts, shell=True)
+            if code != 0:
+                sys.exit(code)
+
+
+def filter_tables_by_date(tables, start_date, end_date):
+    return [
+        (source_table, target_table) for source_table, table_date, target_table in tables
+        if (not start_date or table_date >= start_date) and (not end_date or table_date < end_date)
+    ]
+
+
+def get_dump_load_commands(tables, source_db, target_db, target_host, target_user):
+    dump_opts = ['--data-only', '--no-acl']
+    load_opts = ['-h', target_host, '-U', target_user, target_db]
+    for source_table, target_table in tables:
+        cmd_parts = ['pg_dump', '-t', source_table, source_db] + dump_opts
+        if target_table:
+            cmd_parts += ['|', 'sed', '"s/{}/{}/g"'.format(source_table, target_table)]
+        cmd_parts += ['|', 'psql'] + load_opts
+        yield source_table, target_table, cmd_parts
+
+
+def _confirm(msg):
+    confirm_update = input(msg + ' [yes / no] ')
+    return confirm_update == 'yes'
+
+
+def main():
```

Comment:
why is this not a standard management command (class `Command` with `handle` function as entry point)

Reply:
so that I can be run on the PG node which doesn't have Django installed